Zig.gitignore

```diff
@@ -0,0 +1,7 @@
+# Language: Zig (https://ziglang.org/)
+# Source: https://github.com/ziglang/zig/blob/master/.gitignore (modified)
+
+**/zig-cache/
+**/build/
+**/build-*/
+**/docgen_tmp/
```

Comment:
None of these belong in a generic zig gitignore template, they are cmake/c++ related. Only `zig-cache` and `zig-out` should be included.

Reply:
Maybe also `*.pdb` and `*.exe`?

Reply:
no

Reply:
But Zig creates them when compiling...

Reply:
Are you using the zig build system? I assume not as if you were they would be placed in `zig-out`.

There is no way to know what organization projects not using the zig build system will use, so they should IMO not be handled here.

Reply:
The Zig build system is a project outside of the main compiler, and those who don't use this system should also be taken into account.

Reply:
> The Zig build system is a project outside of the main compiler, 

No, it's part of the main compiler, see the `zig build` command.

> and those who don't use this system should also be taken into account.

There's no way to take them into account, as there is no standard place artifacts are stored if not using the zig build sysetm.

Reply:
Anyhow, I'm not the target audience of this repo. I just add whatever files make sense to my `.gitignore` manually. As such, I don't really care what gets merged here personally. I am however a member of the zig core team.